MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/account/AccountRetrievalService.java

```diff
@@ -0,0 +1,165 @@
+package com.mapbox.mapboxandroiddemo.account;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.preference.PreferenceManager;
+import android.support.annotation.Nullable;
+import android.util.Base64;
+import android.util.Log;
+
+import com.mapbox.mapboxandroiddemo.MainActivity;
+import com.mapbox.mapboxandroiddemo.model.usermodel.UserResponse;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+
+/**
+ * Background service which retrieves Mapbox Account information
+ */
+
+public class AccountRetrievalService extends IntentService {
+
+  private static final String BASE_URL = "https://api.mapbox.com/api/";
+  private static final String ACCESS_TOKEN_URL = "https://api.mapbox.com/oauth/access_token";
+
+  //  TODO: Fill in CLIENT_SECRET
+  private static final String CLIENT_SECRET = "";
+
+  private static String CLIENT_ID;
+  private static String REDIRECT_URI;
+  private String username;
+
+  public AccountRetrievalService() {
+    super("AccountRetrievalService");
+  }
+
+  @Override
+  public void onCreate() {
+    super.onCreate();
+  }
+
+  @Override
+  protected void onHandleIntent(@Nullable Intent intent) {
+    if (intent != null) {
+      String authCode = intent.getStringExtra("AUTHCODE");
+      CLIENT_ID = intent.getStringExtra("CLIENT_ID");
+      REDIRECT_URI = intent.getStringExtra("REDIRECT_URI");
+      getAccessToken(authCode);
+    } else {
+      Log.d("AccountRetrievalService", "onHandleIntent: intent == null");
+    }
+  }
+
+  private void getAccessToken(String code) {
+    OkHttpClient client = new OkHttpClient();
+
+    Request request = new Request.Builder()
+      .addHeader("User-Agent", "Android Dev Preview")
+      .addHeader("Content-Type", "application/x-www-form-urlencoded")
+      .url(ACCESS_TOKEN_URL)
+      .post(RequestBody.create(MediaType.parse("application/x-www-form-urlencoded"),
+        "grant_type=authorization_code&client_id=" + CLIENT_ID + "&client_secret=" + CLIENT_SECRET
+          + "&redirect_uri=" + REDIRECT_URI + "&code=" + code))
+      .build();
+
+    client.newCall(request).enqueue(new okhttp3.Callback() {
```

Comment:
I branched off of this branch. Code above and all of my changes in attempt to switch to retrofit, are on [this branch](https://github.com/mapbox/mapbox-android-demo/tree/ls-switching-last-okhttp3-to-retrofit-for-login)